classes/qtype_questionpy/bridge.php

```diff
@@ -0,0 +1,123 @@
+<?php
+// This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace assignsubmission_qpy\qtype_questionpy;
+
+defined('MOODLE_INTERNAL') || die();
+
+use core\context;
+
+require_once($CFG->dirroot . '/mod/assign/locallib.php');
+
+/**
+ * Bridge class between QuestionPy and assignsubmission_qpy.
+ *
+ * This class is used to retrieve data that is not available through the Question API.
+ *
+ * @package    assignsubmission_qpy
+ * @author     Martin Gauk
+ * @copyright  2025 TU Berlin, innoCampus {@link https://www.questionpy.org}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class bridge extends \qtype_questionpy\question_bridge_base {
+    /** @var \stdClass|null data from assign_submission table. */
+    private $submissiondata = null;
+
+    /**
+     * Get additional LMS attributes.
+     *
+     * @param string[] $requestedattributes
+     * @return string[]
+     */
+    protected function get_additional_lms_attributes(array $requestedattributes): array {
+        $attributes = [];
+
+        if (in_array('attempt_started_at', $requestedattributes)) {
+            $attributes['attempt_started_at'] = ($this->attempt->get_num_steps()) ?
```

Comment:
`get_num_steps()` wollte ich vor Kurzem auch verwenden, allerdings steht im Docstring explizit "_For internal/test code use only._". Alternativ würde das gehen:
```suggestion
            $attributes['attempt_started_at'] = $this->attempt->get_step_iterator()->valid() ?
```
`valid()` greift zwar auch auf `get_num_steps()` zurück, aber dann liegt die Verantwortung nicht bei uns.